Tests/L0/AzureRmWebAppDeployment/_suite.ts

```diff
@@ -0,0 +1,327 @@
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing SPN input', (done) => {
+        
+        setResponseFile('armGood.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('WebAppName', 'mytestapp');
+        tr.setInput('Package', 'webAppPkg.zip');
+        tr.setInput('UseWebDeploy', 'true');
+        
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 0, 'should not have invoked any tool');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Input required: ConnectedServiceName'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr); 
+                assert(tr.failed, 'task should have failed');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing WebAppName', (done) => {
+        
+        setResponseFile('armGood.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('ConnectedServiceName', 'AzureRMSpn');
+        tr.setInput('Package', 'webAppPkg.zip');
+        tr.setInput('UseWebDeploy', 'true');
+        
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 0, 'should not have invoked any tool');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Input required: WebAppName'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr); 
+                assert(tr.failed, 'task should have failed');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing Package Name Input', (done) => {
+        
+        setResponseFile('armGood.json');
+
+        var tr = new trm.TaskRunner('AzureRmWebAppDeployment');
+        tr.setInput('ConnectedServiceName', 'AzureRMSpn');
+        tr.setInput('WebAppName', 'mytestapp');
+        tr.setInput('UseWebDeploy', 'true');
+        
+        tr.run()
+            .then(() => {
+
+                assert(tr.invokedToolCount == 0, 'should not have invoked any tool');
+                assert(tr.stderr.length > 0, 'should have written to stderr');
+                var expectedErr = 'Input required: Package'; 
+                assert(tr.stdErrContained(expectedErr), 'should have said: ' + expectedErr); 
+                assert(tr.failed, 'task should have failed');
+                done();
+
+            })
+            .fail((err) => {
+                done(err);
+            });
+    });
+
+    it('run AzureRmWebAppDeployment Task with missing Package or Folder', (done) => {
```

Comment:
Test cases are being written for missing inputs mostly can you  write few test cases for invalid inputs.